packages/SystemUI/src/com/android/keyguard/KeyguardAbsKeyInputViewController.java

```diff
@@ -215,6 +215,7 @@ protected void verifyPasswordAndUnlock() {
             return;
         }
 
+        mLockPatternUtils.triggerWipeIfDuressPassword(password);
```

Comment:
I think this check has to be performed only if the provided credential is not a valid lockscreen credential, i.e. in  `onChecked()` callback below when `matched` is false.